libs/crypto/merkle-impl/src/main/kotlin/net/corda/crypto/merkle/impl/MerkleProofImpl.kt

```diff
@@ -9,6 +9,15 @@ import net.corda.v5.crypto.merkle.MerkleProofRebuildFailureException
 import net.corda.v5.crypto.merkle.MerkleProofType
 import net.corda.v5.crypto.merkle.MerkleTreeHashDigest
 
+/**
+ * Represent a merkle proof, which shows that some leaf data is in a Merkle tree.
+ *
+ * @param proofType
+ * @param treeSize - total number of leaves in the Merkle tree
+ * @param leaves - list of leaves for which we have data
+ * @param hashes - hashes of each node. Number of elements should match number of elements in leaves.
```

Comment:
I think this is more accurate:
```suggestion
 * @param hashes - hashes of nodes required to rebuild the Merkle tree.
```
And their number can be much lower than the leaves or nodes.
For example, when the left half of a full tree is in the proof, we need only a single hash, the root of the right subtree.

Reply:
Reworked